azafea/event_processors/endless/metrics/machine.py

```diff
@@ -67,3 +73,13 @@ def upsert_machine_live(dbsession: DbSession, machine_id: str) -> None:
                                       set_={'live': True})
 
     dbsession.connection().execute(stmt)
+
+
+def upsert_machine_location(dbsession: DbSession, machine_id: str,
+                            info: Union[Dict[str, Any], List[Any]]) -> None:
+    if isinstance(info, dict):
+        info = {f'location_{key}': value for key, value in info.items()}
+        stmt = insert(Machine.__table__).values(machine_id=machine_id, **info)
```

Comment:
As above, there's no guarantee what keys the dictionary will contain. I can put an arbitrary key `foo` into `/etc/metrics/location.conf` on my system, and it will cause this query to fail since there is no `location_foo` column in the table.

Reply:
You’re right, I thought that the keys were fixed.

Reply:
Done.